x/valset-pref/validator_set.go

```diff
@@ -286,6 +288,57 @@ func (k Keeper) withdrawExistingValSetStakingPosition(ctx sdk.Context, delegator
 	return nil
 }
 
+// ForceUnlockBondedOsmo allows breaking of a bonded lockup (by ID) of osmo, of length <= 2 weeks.
+// We want to later have osmo incentives get auto-staked, we want people w/ no staking positions to
+// get their osmo auto-locked. This function takes all that osmo and stakes according to your
+// current validator set preference.
+// (Note: Noting that there is an implicit valset preference if you've already staked)
+func (k Keeper) ForceUnlockBondedOsmo(ctx sdk.Context, lockID uint64, delegatorAddr string) (sdk.Coin, error) {
+	// Checks if sender is lock ID owner
+	lock, err := k.lockupKeeper.GetLockByID(ctx, lockID)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+	if lock.GetOwner() != delegatorAddr {
+		return sdk.Coin{}, fmt.Errorf("delegator (%s) and lock owner (%s) does not match", delegatorAddr, lock.Owner)
+	}
+
+	unlockedOsmoAmount := sdk.NewInt(0)
+
+	// check if lock contains osmo tokens
+	for _, lockToken := range lock.Coins {
+		if lockToken.Denom == appParams.BaseCoinUnit {
+			unlockedOsmoAmount = unlockedOsmoAmount.Add(lockToken.Amount)
+		}
+	}
```

Comment:
Wait, what if the lock contains osmo and some other asset. Wouldn't this still allow us to force unlock it (which i think we need to explicitly not allow right?)

Reply:
hmm i thought all the LP shares had 2 assets which we eventually lock. Can we have single asset locks? 

My thought process here was to unlock the lock(regardless of 2 assets)  and only use the uosmo shares to delegate to a valset

Reply:
Well I think a majority of lock are single asset locks right? (they lock gamm shares which is a single asset). A lock AFAIK can consist of either a single asset or multiple assets.

I think the intent of this is to have osmo rewards go straight to lockup which would be a single asset lock. But we wouldn't want someone to be able to abuse this by being able to force unlock a lock just because it has osmo in it, does that make sense?

Reply:
I understand the gamm shares lock, since they are LP shares we view them as single asset. But i'm still a lil confused as to why would only want to lock osmo. I always thought the flow was; 
`provide liquidity in pool (2asset) -> get LP shares (1asset) -> lock LP shares`

Reply:
resolving this because of new knowledge i received from our call